vignettes/markdown.Rmd

```diff
@@ -141,33 +148,52 @@ URLs inside angle brackets are also automatically converted to hyperlinks:
 #' The main R web site is at <https://r-project.org>.
 ```
 
-Markdown notation can be used to create links to other manual pages. There are six kinds of links:
-
-1. Link to another function in the same package: `[func()]`. These
-   links will be typeset as code, and they are equavalent to
-   `\code{\link[=func]{func()}`.
-2. Link to a (non-function) object, class, data set, etc. in the same
-   same package: `[object]`. These links that *not* typeset as code,
-   so if you want them as code, enclose them in backticks (inside the
-   brackets).
-3. Link to a function from another package: `[pkg::func()]`. These links
-   will be typeset as code.
-4. Link to a (non-function) object in another package: `[pkg::object]`.
-   These links will not be typeset as code.
-5. Link to an object in the same package, with a different link text:
-   `[link text][object]`. Here `object` can be a function, but the link
-   text is not typeset as code.
-6. Link to an object in another package, with different link text:
-   `[link text][pkg::object]`. This is not typeset as code.
-
-S3/S4 classes can be linked the same way:
-
-```r
-#' * [terms][terms.object] becomes \link[=terms.object]{terms}
-#' * [abc][abc-class] becomes \link[=abc-class]{abc}
-```
-
-Or a shorthand notation can be used: `[abc-class]` is converted to `\linkS4class{abc}` and `[pkg::abc-class]` is converted to `\link[pkg:abc-class]{pkg::abc}`.
+Markdown notation can also be used to create links to other help topics. There are two basic forms:
+
+  * `[ref]`: The target topic and the link text are the same.
+  * `[text][ref]`: Link text differs from the target.
+
+with variations depending on whether `ref` is in the same package and whether the link is automatically typeset as code. Use backticks inside the square brackets to typeset the link text as code, when it would not be otherwise:
+
```

Comment:
The rule is that function names are always typeset as code, i.e. anything that ends with `()`. Otherwise you might want to typeset as code, might not.  For `[text][ref]`maybe not, since the `text` is might not be an object name, just generic text. So I think ``[`text`][ref]`` below is a bit misleading. 

TBH I am not even sure if we should typeset class names as code, maybe.  The style guide does not say anything about that.  I often just write 
```r
#' ... a [processx:process] object ...
```
If I want this as code, I think that would be
```r
#' ... a [`processx:process`][processx::process] object ...
```
which is quite cumbersome, and if we decide that class names should be code, we should probably have some shorthand for this.

Reply:
> The rule is that function names are always typeset as code, i.e. anything that ends with ()

So is it based on the presence of `()` or actually determining if the target is a function? Now that I say this, I supposed it's probably just based on the `()`.

I got into the cumbersome situation you're talking about with `data.frame` and `data.frame()` in tibble. There *is* no topic for the `data.frame` class. So when you talk about the class, it seems you have to link to the `data.frame()` constructor function. But I think "data.frame" would look weird if it were *not* typeset as code.

Reply:
Yeah but there *could* and arguably *should* be a topic called "data.frame-class", like [httr::Token-class](https://www.rdocumentation.org/packages/httr/versions/1.3.1/topics/Token-class), [Matrix-class](https://www.rdocumentation.org/packages/Matrix/versions/1.2-14/topics/Matrix-class), etc.